library/std/src/io/error.rs

```diff
@@ -156,6 +191,62 @@ pub enum ErrorKind {
     /// [`Ok(0)`]: Ok
     #[stable(feature = "rust1", since = "1.0.0")]
     WriteZero,
+    /// The underlying storage (typically, a filesystem) is full.
+    ///
+    /// This does not include out of quota errors.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    NoStorageSpace,
+    /// Invalid seek
+    ///
+    /// Seeking was attempted on an open file handle which is not suitable for seeking - for
+    /// example, on Unix, a named pipe opened with `File::new`.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    InvalidSeek,
+    /// Filesystem quota was exceeded.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    FilesystemQuotaExceeded,
+    /// File larger than allowed or supported.
+    ///
+    /// This might arise from a hard limit of the underlying filesystem or file access API, or from
+    /// an administratively imposed resource limitation.  Simple disk full, and out of quota, have
+    /// their own errors.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    FileTooLarge,
+    /// Resource is busy.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    ResourceBusy,
```

Comment:
Not sure, this may be [`ERROR_NETWORK_BUSY`](https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes--0-499-), [`ERROR_BUSY_DRIVE`](https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes--0-499-), [`ERROR_PATH_BUSY`](https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes--0-499-), [`ERROR_BUSY`](https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes--0-499-), [`ERROR_PIPE_BUSY`](https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes--0-499-)

Reply:
Of those I think only `ERROR_BUSY` is right.  The others are all things that can't happen on Unix.  I'm inclined to keep different errors separate as much as possible since that really helps sysadmins with debugging and it can also help programmers who need to deal with "interesting" edge cases.